Добавлено решение ДЗ №1

karelalex opened the pull request.

Commit:
Добавлено решение ДЗ №1

vlad-iv commented on hw01/pom.xml:
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Можно добавить корневой pom
где модулями будут ДЗ
пример
https://github.com/OtusTeam/Spring/blob/master/2024-09/spring-01/pom.xml

karelalex replied:
Добавил, и перенёс туда настройку чекстайл плагина, чтобы он стал сразу на всё.

vlad-iv commented on hw01/pom.xml:
```diff
@@ -0,0 +1,134 @@
+    <artifactId>hw01-xml-config</artifactId>
+    <version>1.0</version>
+
+    <properties>
```
Предлагаю для проверки сборки проекта на гитхабе
создать файл .github/workflows/build.yml
Пример:
https://github.com/OtusTeam/Spring/blob/master/.github/workflows/build.yml

karelalex replied:
Добавил

vlad-iv commented on hw01/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+            var question = questions.get(i);
+            ioService.printFormattedLine("%d. %s",i + 1, question.text());
+            var answers = question.answers();
+            answers.forEach(answer -> ioService.printFormattedLine("\u0009 %s", answer.text()));
```
Предлагаю перед каждым вариантом ответа указать его номер
с нумерацией от 1
(в следующих ДЗ потребуется, что бы пользователь ввел этот номер ответа)

karelalex replied:
Поправил

vlad-iv commented on hw01/src/test/java/ru/otus/hw/service/TestRunnerServiceImplTest.java:
```diff
@@ -0,0 +1,24 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TestRunnerServiceImplTest {
```
Можно добавить тесты на CsvQuestionDao 
проверить что вопросы верно загружаются, а если указан не верный ресурс то будет исключение

karelalex replied:
Это очень похоже на одно из заданий ДЗ 2, но добавил, пусть будет и тут.

Commit:
Исправления замечаний по ДЗ №1

Добавлен общий POM
Добавлен файл настройки workflow для проверки сборки на GitHub
Добавлен тест CsvQuestionDao
Ответы выводятся с номерами
